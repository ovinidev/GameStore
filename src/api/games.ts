import { Game } from "@interfaces/games";
import axios from "axios";

export const axiosInstance = axios.create({
	baseURL:
		"https://gist.githubusercontent.com/ovinidev/62f6a75e0fb6786584caaead7846ee14/raw/games.json",
});

export const getGamesFromApi = async () => {
	const { data } = await axiosInstance.get<Game[]>("");

	return data;
};

export const getGamesByIdFromApi = async (id: number) => {
	const game = await getGamesFromApi();

	const gameById = game.find((game) => game.id === id);

	if (!gameById) {
		throw new Error("Cart not found");
	}

	return gameById;
};
