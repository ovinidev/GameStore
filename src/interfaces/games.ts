export interface Game {
	id: number;
	title: string;
	platform: string;
	genre: string;
	release_date: string;
	image: string;
	price: number;
	description: string;
}
