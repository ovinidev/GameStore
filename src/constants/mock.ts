import { Game } from "@interfaces/games";

export const GAME_MOCK: Game = {
	title: "Game Beta",
	description: "A thrilling game",
	genre: "Action",
	id: 1,
	image: "image-url",
	platform: "PC",
	price: 60,
	release_date: "2024-08-01",
};

export const GAMES_MOCK: Game[] = [
	{
		title: "Game Beta",
		description: "A thrilling game",
		genre: "Action",
		id: 1,
		image: "image-url",
		platform: "PC",
		price: 60,
		release_date: "2024-08-01",
	},
	{
		title: "Game Alpha",
		description: "A thrilling game",
		genre: "Action",
		id: 2,
		image: "image-url",
		platform: "PC",
		price: 60,
		release_date: "2024-08-01",
	},
];

export const CART_MOCK: Game[] = [
	{
		title: "Game Beta",
		description: "A thrilling game",
		genre: "Action",
		id: 1,
		image: "image-url",
		platform: "PC",
		price: 60,
		release_date: "2024-08-01",
	},
	{
		title: "Game Alpha",
		description: "A thrilling game",
		genre: "Action",
		id: 2,
		image: "image-url",
		platform: "PC",
		price: 60,
		release_date: "2024-08-01",
	},
];
