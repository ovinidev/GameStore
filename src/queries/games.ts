import { getGames } from "@api/games";
import { GAMES } from "@constants/entities";
import { useQuery } from "@tanstack/react-query";

export const useGames = () => {
	return useQuery({
		queryKey: [GAMES],
		queryFn: getGames,
	});
};
