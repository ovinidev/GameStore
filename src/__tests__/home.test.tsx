import React from "react";
import { render } from "@testing-library/react-native";
import Home from "../app/index";
import { useGames } from "@queries/games";
import { GAMES_MOCK } from "@constants/mock";

jest.mock("expo-router", () => ({
	Stack: {
		Screen: "Screen",
		Navigator: "Navigator",
	},
	useRouter: jest.fn().mockReturnValue({
		push: jest.fn(),
		setParams: jest.fn(),
	}),
}));

jest.mock("@queries/games", () => ({
	useGames: jest.fn(),
}));

jest.mock("@hooks/useCart", () => ({
	useCart: jest.fn().mockReturnValue({
		data: jest.fn(),
	}),
}));

const mockUseGames = useGames as jest.Mock;

describe("Home", () => {
	it("should match the snapshot", () => {
		mockUseGames.mockReturnValue({
			data: GAMES_MOCK,
		});

		const { toJSON } = render(<Home />);
		expect(toJSON()).toMatchSnapshot();
	});
});
