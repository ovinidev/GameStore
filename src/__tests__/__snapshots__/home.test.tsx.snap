// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page should match the snapshot 1`] = `
<View
  className="flex flex-1 items-center bg-slate-950"
>
  <Screen
    options={
      {
        "headerRight": [Function],
        "headerTitle": [Function],
      }
    }
  />
  <RCTScrollView
    ItemSeparatorComponent={[Function]}
    data={
      [
        {
          "description": "A thrilling game",
          "genre": "Action",
          "id": 1,
          "image": "image-url",
          "platform": "PC",
          "price": 60,
          "release_date": "2024-08-01",
          "title": "Game Beta",
        },
        {
          "description": "A thrilling game",
          "genre": "Action",
          "id": 2,
          "image": "image-url",
          "platform": "PC",
          "price": 60,
          "release_date": "2024-08-01",
          "title": "Game Alpha",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          style={
            {
              "flexDirection": "row",
            }
          }
        >
          <View
            accessibilityLabel="Adicionar ao carrinho"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Image
              className="h-[170] w-full"
              resizeMode="cover"
              source={
                {
                  "uri": "image-url",
                }
              }
            />
            <Text
              className="py-2 text-center font-semibold text-md text-slate-50"
            >
              Game Beta
            </Text>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                className="text-center font-bold text-md text-slate-50"
              >
                Adicionar
              </Text>
            </View>
          </View>
          <View
            accessibilityLabel="Adicionar ao carrinho"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "opacity": 1,
              }
            }
          >
            <Image
              className="h-[170] w-full"
              resizeMode="cover"
              source={
                {
                  "uri": "image-url",
                }
              }
            />
            <Text
              className="py-2 text-center font-semibold text-md text-slate-50"
            >
              Game Alpha
            </Text>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "opacity": 1,
                }
              }
            >
              <Text
                className="text-center font-bold text-md text-slate-50"
              >
                Adicionar
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
